.github/workflows/build.yml

```diff
@@ -16,67 +16,34 @@ on:
       - develop
 
 jobs:
-  build:
+  test_and_build:
     if: github.event.pull_request.draft == false
     runs-on: ${{ matrix.os }}
     strategy:
       fail-fast: false
       matrix:
         os: ["ubuntu-latest"]
-        python-version: ["3.10"]
+        python-version: ["3.10", "3.11", "3.12", "3.13"]
```

Comment:
For environmental reasons, I am not a big fan of running the CI on so many python versions (see [blog](https://blog.esciencecenter.nl/reduce-reuse-recycle-save-the-planet-one-github-action-at-a-time-4ab602255c3f)), given that at the end of the day, it's unlikely that it fails in only one.
While it's nice to offer the flexibility, I don't see much harm to instead state that we officially support any one python version but that it likely works on newer versions as well.

Reply:
Good suggestion! I agree that this is not very environmentally friendly. There are a few alternatives that reduce the number of runs while still ensuring it'll run on newer versions of Python. 

- Only run on the earliest and latest supported Python release. Is it safe to assume it will run on all intermittent versions?
- Stop official support for 3.10 and 3.11. Both only receive security fixes and should not be the base for new projects. Aggressive removal of supported versions means existing projects can only use new features if they update (assuming something actually breaks) or wing it (if their workflow still works). 
- Remove tests for all versions from the test_and_build action, but add it to the Draft Github Release action. This makes sure we don't release anything that breaks, but does not test every PR while it is being developed.

Reply:
> Remove tests for all versions from the test_and_build action, but add it to the Draft Github Release action. This makes sure we don't release anything that breaks, but does not test every PR while it is being developed.

I really like this option, actually. Most breaks will be picked up in all versions anyway, and the few that are version specific can be picked up like this. The only thing to make sure of is to give a clear "error message" when it fails, because that could otherwise be very confusing.